### PR TITLE
Add the type information and comments to functions

### PR DESCRIPTION
**Pull request by peterBrxwn**

### CODE REVIEW

**Comment by peterBrxwn** [comment]
[[RN] Add the type information and comments to functions](https://trello.com/c/TBjop3qT/881-rn-add-the-type-information-and-comments-to-functions)

**Comment by turtledreams** [comment]
Run linter here. Remove comments. Remove the import at top.

File: Countly.d.ts [hunk 1]
```diff
@@ -0,0 +1,824 @@
...
+    endTime: number;
+  }
+
+  type ValidationFunction = (
+    stringValue: string,
+    stringName: string,
+    functionName: string
+  ) => Promise<string | null>;
+
+  type ResultObject = (
```
Comment (turtledreams) [inline Countly.d.ts:57]: I checked this at the example we have and it was giving an error. Changing it to this fixed it:
  type ResultObject = {
    error: string,
    data: object,
  };
So can you go through the example app and check the errors coming through and fix them.

**Review by turtledreams — CHANGES_REQUESTED** [review]

File: Countly.d.ts [hunk 2]
```diff
@@ -0,0 +1,830 @@
...
+// export function getAvailableFeedbackWidgets(): Promise<FeedbackWidget[]>;
+// export function presentFeedbackWidget(widgetType: string, widgetId: string, closeButtonText: string): void;
+// export function logError(tag: string, message: string): void;
+// export function validateString(value: string, label: string, context: string): string | Promise<string>;
+// export function setViewTracking(boolean: boolean): Promise<void>;
+// export function startTrace(traceKey: string): void;
+// export function cancelTrace(traceKey: string): void;
+// export function clearAllTraces(): void;
+// export function endTrace(traceKey: string, customMetric?: TraceCustomMetric): void;
+// export function recordNetworkTrace(data: NetworkTraceData): void;
```
[the remark this answers is not in the document]
Comment (peterBrxwn) [reply Countly.d.ts]: yup... fixed

**Review by turtledreams — CHANGES_REQUESTED** [review]
move types outside of declaration

**Comment by turtledreams** [comment]
dmed u an image

File: Countly.d.ts
Comment (ijunaid) [inline Countly.d.ts:1]: @peterBrxwn, have you tested it? 
Do we have an example file/app for typescript to check that it's working fine?

**Review by ijunaid — APPROVED** [review]
Overall, it looks good to me. Please make sure on your side that it's working fine with typescript apps, and if we have an example app/file for typescript, that would be more helpful.

File: Countly.d.ts [hunk 3]
```diff
@@ -0,0 +1,875 @@
...
+    export function cancelEvent(eventName: string): string | void;
+
+    /**
+     *
+     * End Event
+     *
+     * @param {string | object} options event options
+     * @return {string | void} error message or void
+     */
+    export function endEvent(options: CountlyEventOptions): string | void;
```
Comment (turtledreams) [inline Countly.d.ts:447]: this can also take string

**Review by turtledreams — CHANGES_REQUESTED** [review]

**Pull request merged**